### PR TITLE
fix: suppport apiendpoints using the cdn

### PR DESCRIPTION
**Pull request by MarcMcIntosh**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/core/__tests__/src/models.test.ts [hunk 1]
```diff
@@ -86,4 +86,21 @@ describe("Manifest", () => {
     );
     expect(decoded).toBeNull();
   });
+
+  test("apiEnpoint: https://foo-bar.cdn.prismic.io/api/v2", () => {
+    const input = { apiEndpoint: "https://foo-bar.cdn.prismic.io/api/v2" };
+    const result = Manifest.decode(input);
+    expect(isRight(result)).toBeTruthy();
+    const decoded = pipe(
```
Comment (Heyrows) [inline packages/core/__tests__/src/models.test.ts:94]: If the test before says "isRight" = true, I think you can do result.right.
Like the either in scala

File: packages/core/src/models/Manifest.ts [hunk 2]
```diff
@@ -9,8 +9,8 @@ class ApiEndPointType extends t.Type<string> {
       (input, context) => {
         if (typeof input !== "string") return t.failure(input, context);
         const regx = new RegExp(
-          "^https?://[a-z0-9][a-z0-9-]{2,}[a-z0-9].(prismic.io|wroom.io|wroom.test)/api/v2/?$",
-          "i"
+          "^https?://[a-z0-9][a-z0-9-]{2,}[a-z0-9](.cdn)?.(prismic.io|wroom.io|wroom.test)/api/v2/?$",
```
Comment (Heyrows) [inline packages/core/src/models/Manifest.ts:12]: I've tested the Regexp online and it seems that the regexp now returns 2 groups: `cdn` and `prismic.io`.
How do you know that your function will return the second parameter now ?

Comment (MarcMcIntosh) [reply packages/core/src/models/Manifest.ts:12]: it users the the `test` method, it only returns true or false

File: packages/core/__tests__/src/models.test.ts (hunk 1 above)
Comment (MarcMcIntosh) [reply packages/core/__tests__/src/models.test.ts:94]: That's not how io-ts works,

it returns an object with a _tag prop that's either "right" or "left"

Comment (MarcMcIntosh) [reply packages/core/__tests__/src/models.test.ts:94]: https://github.com/gcanti/fp-ts/blob/master/src/internal.ts#L33

**Review by Heyrows — APPROVED** [review]

**Pull request merged**